packages/vector_graphics_compiler/lib/src/svg/masking_optimizer.dart

```diff
@@ -0,0 +1,362 @@
+import 'package:vector_graphics_compiler/vector_graphics_compiler.dart';
+import 'package:vector_graphics_compiler/src/svg/path_ops.dart' as path_ops;
+
+class _EmptyNode extends Node {
+  const _EmptyNode();
+
+  @override
+  S accept<S, V>(Visitor<S, V> visitor, V data) {
+    return visitor.visitEmptyNode(this, data);
+  }
+
+  @override
+  void visitChildren(NodeCallback visitor) {}
+}
+
+class _Result {
+  _Result(this.node);
+
+  final Node node;
+  final List<Node> children = <Node>[];
+  Node parent = const _EmptyNode();
+  bool deleteMaskNode = true;
+}
+
+/// Converts a vector_graphics PathFillType to a path_ops FillType.
+path_ops.FillType toPathOpsFillTyle(PathFillType fill) {
+  switch (fill) {
+    case PathFillType.evenOdd:
+      {
+        return path_ops.FillType.evenOdd;
+      }
+
+    case PathFillType.nonZero:
+      {
+        return path_ops.FillType.nonZero;
+      }
+  }
+}
+
+/// Converts a path_ops FillType to a vector_graphics PathFillType
+PathFillType toVectorGraphicsFillType(path_ops.FillType fill) {
+  switch (fill) {
+    case path_ops.FillType.evenOdd:
+      {
+        return PathFillType.evenOdd;
+      }
+
+    case path_ops.FillType.nonZero:
+      {
+        return PathFillType.nonZero;
+      }
+  }
+}
+
+/// Converts vector_graphics Path to path_ops Path.
+path_ops.Path toPathOpsPath(Path path) {
+  final path_ops.Path newPath = path_ops.Path(toPathOpsFillTyle(path.fillType));
+
+  for (PathCommand command in path.commands) {
+    switch (command.type) {
+      case PathCommandType.line:
+        {
+          final LineToCommand lineToCommand = command as LineToCommand;
+          newPath.lineTo(lineToCommand.x, lineToCommand.y);
+        }
+        break;
+      case PathCommandType.cubic:
+        {
+          final CubicToCommand cubicToCommand = command as CubicToCommand;
+          newPath.cubicTo(
+              cubicToCommand.x1,
+              cubicToCommand.y1,
+              cubicToCommand.x2,
+              cubicToCommand.y2,
+              cubicToCommand.x3,
+              cubicToCommand.y3);
+        }
+        break;
+      case PathCommandType.move:
+        {
+          final MoveToCommand moveToCommand = command as MoveToCommand;
+          newPath.moveTo(moveToCommand.x, moveToCommand.y);
+        }
+        break;
+      case PathCommandType.close:
+        {
+          newPath.close();
+        }
+        break;
+    }
+  }
+
+  return newPath;
+}
+
+/// Converts path_ops Path to VectorGraphicsPath.
+Path toVectorGraphicsPath(path_ops.Path path) {
+  final List<PathCommand> newCommands = <PathCommand>[];
+
+  int index = 0;
+  final Float32List points = path.points;
+  for (path_ops.PathVerb verb in path.verbs.toList()) {
+    switch (verb) {
+      case path_ops.PathVerb.moveTo:
+        {
+          newCommands.add(MoveToCommand(points[index++], points[index++]));
+        }
+        break;
+      case path_ops.PathVerb.lineTo:
```

Comment:
Here and elsewhere, switch cases don't need their own `{` `}` blocks, since they are already a new scope. That isn't true in some other languages.

```suggestion
    switch (verb) {
      case path_ops.PathVerb.moveTo:
          newCommands.add(MoveToCommand(points[index++], points[index++]));
        break;
      case path_ops.PathVerb.lineTo:
```

Reply:
fixed.